### ISSUE TITLE
Error detected while processing function SnipRunLauncher

### ISSUE THREAD

**Issue by chenhui2016** [michaelb/sniprun#265]
Ubuntu 22.04 

Nvim 0.9.4

![image](https://github.com/michaelb/sniprun/assets/66300288/91ffc104-39a1-4460-8375-2b05ef580e1e)

**Comment by michaelb** [comment]
Hi thanks for submitting an issue about this problem.... However I don't have enough data to understand what's going on (and I'm quite unable to reproduce). My ideas for now are:

- What does `:checkhealth sniprun` says ?
- did you customize sniprun's `binary_path`
- does an executable exist at the path mentionned in the error (and what happens if you launch it ?)

**Comment by chenhui2016** [comment]
Thanks for your reply!
I just want to run c language using sniprun, I installed sniprun correctly and using the default configuration.
1: `:checkhealth sniprun` say
![image](https://github.com/michaelb/sniprun/assets/66300288/9ae39c40-85d7-426b-86e6-46c6a4c57217)
2:I'm not customize sniprun's `binary_path`
3.I just used the default installation method and installed sniprun correctly.
![image](https://github.com/michaelb/sniprun/assets/66300288/19a5e4fe-0127-4ea1-bcf2-21b2b8205426)

**Comment by michaelb** [comment]
The checkhealth suggests something very important (the sniprun binary) doesn't exist

There should be no reason the binary is not present, save for internet connection strangeness (unlikely), or an inoportune neovim crash/close while the binary wasn't completely downloaded.

In both case I would try to reinstall the plugin, but if it doesn't work, I'd find where the plugin manager (lazy.nvim in your case) put sniprun, check the./target/release/sniprun binary and try to run the ./install.sh script

**Comment by chenhui2016** [comment]
Thanks for your reply！
Fine , The reason is Internet connetction!
![image](https://github.com/michaelb/sniprun/assets/66300288/9bb8e9c9-d312-4275-8585-bdf065c2a4a9)

But, if I manually download a binary file, where should I place it? Is it this address /home/ch/.local/share/nvim/lazy/sniprun?

**Comment by chenhui2016** [comment]
OK， I have solved. The last question, I don’t know if it’s a bug or if it will have any impact in the future.
1.When I manage the plugin in the lazy interface I run checkhealth and everything is correct.
2. When I finished running the cpp file and got the correct results, then I ran the checkhealth command, the following error occurred:
![image](https://github.com/michaelb/sniprun/assets/66300288/ea36c2da-675c-433a-bcb1-aedbab6b7e13)
4. Is this a bug? And I made the changes and reinstalled sniprun according to its error report, and it still shows this error.
5. Thank you very much for making this plug-in. Thank you very much.
